Composer/packages/lib/code-editor/src/lu/DefineEntityButton.tsx

```diff
@@ -48,20 +52,47 @@ const getCommandBarButton = (tooltipContent: string) =>
   withTooltip({ content: tooltipContent }, DefaultCommandBarButton);
 
 type Props = {
-  onDefineEntity: (entityType: ToolbarLuEntityType, entityName?: string) => void;
+  onDefineEntity: (
+    entityType: ToolbarLuEntityType,
+    data: Partial<{ entityName: string; entityDefinition: string }>
+  ) => void;
   disabled?: boolean;
   tooltip?: string;
+  luFile?: LuFile;
 };
 
 export const DefineEntityButton = React.memo((props: Props) => {
-  const { onDefineEntity, disabled = false, tooltip } = props;
+  const { luFile, onDefineEntity, disabled = false, tooltip } = props;
 
+  const entities = useLuEntities(luFile);
+
+  const [showListEntityCreationDialog, setShowListEntityCreationDialog] = React.useState(false);
   const { iconName, text } = React.useMemo(() => getLuToolbarItemTextAndIcon('defineEntity'), []);
   const { onRenderMenuList, query, setQuery } = useSearchableMenuListCallback(
     formatMessage('Search prebuilt entities')
   );
+
   const noSearchResultsMenuItem = useNoSearchResultMenuItem(formatMessage('no prebuilt entities found'));
 
+  const onClickEntityType = React.useCallback(
+    (entityType: ToolbarLuEntityType, ignoreEntityTypeList: ToolbarLuEntityType[] = [], entityName?: string) => {
+      if (ignoreEntityTypeList.includes(entityType)) return;
+
+      switch (entityType) {
+        case 'list':
+          setShowListEntityCreationDialog(true);
+          break;
+        case 'ml':
+        case 'prebuilt':
+          onDefineEntity(entityType, entityName ? { entityName } : {});
+          break;
+        default:
+          throw `${entityType} is not supported!`;
```

Comment:
formatMessage

Reply:
it's a throw message, wont surface to UX